### PR TITLE
provide a validity check to prevent against Integer overflow conditions

### PR DESCRIPTION
**Pull request by quangnh89**

### CODE REVIEW

**Comment by aquynh** [comment]
@tandasat, can you ack?

File: windows/winkernel_mm.c [hunk 1]
```diff
@@ -33,8 +34,15 @@ void * CAPSTONE_API cs_winkernel_malloc(size_t size)
 
 	// FP; a use of NonPagedPool is required for Windows 7 support
 #pragma prefast(suppress : 30030)		// Allocating executable POOL_TYPE memory
-	CS_WINKERNEL_MEMBLOCK *block = (CS_WINKERNEL_MEMBLOCK *)ExAllocatePoolWithTag(
-			NonPagedPool, size + sizeof(CS_WINKERNEL_MEMBLOCK), CS_WINKERNEL_POOL_TAG);
+	size_t numberOfBytes = 0;
```
Comment (tandasat) [inline windows/winkernel_mm.c]: can you make it snake_case to in line with other Capstone code?

File: windows/winkernel_mm.c [hunk 2]
```diff
@@ -33,8 +34,15 @@ void * CAPSTONE_API cs_winkernel_malloc(size_t size)
...
+	CS_WINKERNEL_MEMBLOCK *block = NULL;
+	// A specially crafted size value can trigger the overflow.
+	// If the sum in a value that overflows or underflows the capacity of the type,
+	// the function returns NULL.
+	if (!NT_SUCCESS(RtlSizeTAdd(size, sizeof(CS_WINKERNEL_MEMBLOCK), &numberOfBytes)))
+		return NULL;
```
Comment (tandasat) [inline windows/winkernel_mm.c]: Can you put this inside `{}` like the below `if (!block) {` to in line with other Capstone code?

Comment (quangnh89) [inline windows/winkernel_mm.c]: yes. I am editing. :)

**Comment by tandasat** [comment]
Good catch. Commented on minor style issues.

**Comment by tandasat** [comment]
Looks good to me. Thank you for the PR.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**